Compiling error on OSX. Missing readline

My environment:
Mac OS 10.10
readline 6.3.8 installed from `brew` located at `/usr/local/Cellar/readline/6.3.8`

```
gcc -std=c99 -I. -I../include -I../common -I/opt/local/include -I../liblua -I/usr/local/Cellar/readline/6.3.6/include/ -Wall  -g -O4 -DHAVE_GUI -c -o obj/ui.o ui.c
clang: warning: -O4 is equivalent to -O3
ui.c:48:19: error: use of undeclared identifier 'rl_readline_state'
        int need_hack = (rl_readline_state & RL_STATE_READCMD) > 0;
                         ^
ui.c:48:39: error: use of undeclared identifier 'RL_STATE_READCMD'
        int need_hack = (rl_readline_state & RL_STATE_READCMD) > 0;
                                             ^
ui.c:52:16: warning: implicit declaration of function 'rl_copy_text' is invalid in C99 [-Wimplicit-function-declaration]
                saved_line = rl_copy_text(0, rl_end);
                             ^
ui.c:52:14: warning: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
                saved_line = rl_copy_text(0, rl_end);
                           ^ ~~~~~~~~~~~~~~~~~~~~~~~
ui.c:53:3: warning: implicit declaration of function 'rl_save_prompt' is invalid in C99 [-Wimplicit-function-declaration]
                rl_save_prompt();
                ^
ui.c:54:3: warning: implicit declaration of function 'rl_replace_line' is invalid in C99 [-Wimplicit-function-declaration]
                rl_replace_line("", 0);
                ^
ui.c:66:3: warning: implicit declaration of function 'rl_restore_prompt' is invalid in C99 [-Wimplicit-function-declaration]
                rl_restore_prompt();
                ^
5 warnings and 2 errors generated.
make[1]: *** [obj/ui.o] Error 1
make: *** [client/all] Error 2
```

I fix this error by adding readline path to `proxmark3/client/Makefile`

```
LDLIBS = -L/usr/local/Cellar/readline/6.3.8/lib/ -L/opt/local/lib -L/usr/local/lib ../liblua/liblua.a -lreadline -lpthread -lm
CFLAGS = -std=c99 -I. -I../include -I../common -I/opt/local/include -I../liblua -I/usr/local/Cellar/readline/6.3.8/include/ -Wall $(COMMON_FLAGS) -g -O4
```

--- a/COMPILING.txt
+++ b/COMPILING.txt
@@ -109,7 +109,7 @@ Tested on OSX 10.10 Yosemite
 1 - Install Xcode and Xcode Command Line Tools
 
 2 - Install Homebrew and dependencies
-    brew install readline libusb p7zip libusb-compat wget qt5 pkgconfig
+    brew install libusb p7zip libusb-compat wget qt5 pkgconfig
 
 3 - Install DevKitARM for OSX
     Option 1:
@@ -121,8 +121,8 @@ Tested on OSX 10.10 Yosemite
 
 4 - Edit proxmark3/client/Makefile adding path to readline and qt5
 
-    LDLIBS = -L/usr/local/opt/readline/lib -L/usr/local/opt/qt5/lib -L/opt/local/lib -L/usr/local/lib ../liblua/liblua.a -lreadline -lpthread -lm
-    CFLAGS = -std=c99 -I/usr/local/opt/qt5/include -I/usr/local/opt/readline/include -I. -I../include -I../common -I../zlib -I/opt/local/include -I../liblua -Wall $(COMMON_FLAGS) -g -O4
+    LDLIBS = -L/usr/local/opt/qt5/lib -L/opt/local/lib -L/usr/local/lib ../liblua/liblua.a -lreadline -lpthread -lm
+    CFLAGS = -std=c99 -I/usr/local/opt/qt5/include -I. -I../include -I../common -I../zlib -I/opt/local/include -I../liblua -Wall $(COMMON_FLAGS) -g -O4
 
     If your old brew intallation use /usr/local/Cellar/ path replace /usr/local/opt/readline/lib with your actuall readline and qt5 path. See homebrew manuals.
 

--- a/COMPILING.txt
+++ b/COMPILING.txt
@@ -3,7 +3,7 @@ The project compiles on Linux, Mac OS X and Windows (MinGW/MSYS).
 it requires:
 - gcc >= 4.4
 - libpthread
-- libreadline
+- GNU libreadline or BSD libedit (editline)
 - libusb
 - perl
 - an ARM cross-compiler to compile the firmware
@@ -119,12 +119,10 @@ Tested on OSX 10.10 Yosemite
         brew tap nitsky/stm32
         brew install arm-none-eabi-gcc
 
-4 - Edit proxmark3/client/Makefile adding path to readline and qt5
+4 - Edit proxmark3/client/Makefile adding path qt5
 
     LDLIBS = -L/usr/local/opt/qt5/lib -L/opt/local/lib -L/usr/local/lib ../liblua/liblua.a -lreadline -lpthread -lm
     CFLAGS = -std=c99 -I/usr/local/opt/qt5/include -I. -I../include -I../common -I../zlib -I/opt/local/include -I../liblua -Wall $(COMMON_FLAGS) -g -O4
-
-    If your old brew intallation use /usr/local/Cellar/ path replace /usr/local/opt/readline/lib with your actuall readline and qt5 path. See homebrew manuals.
 
 5 - Set Environment
 

--- a/client/ui.c
+++ b/client/ui.c
@@ -48,7 +48,9 @@ void PrintAndLog(char *fmt, ...)
 			logging=0;
 		}
 	}
-	
+
+#ifdef RL_STATE_READCMD
+	// We are using GNU readline.
 	int need_hack = (rl_readline_state & RL_STATE_READCMD) > 0;
 
 	if (need_hack) {
@@ -58,6 +60,10 @@ void PrintAndLog(char *fmt, ...)
 		rl_replace_line("", 0);
 		rl_redisplay();
 	}
+#else
+	// We are using libedit (OSX), which doesn't support this flag.
+	int need_hack = 0;
+#endif
 	
 	va_start(argptr, fmt);
 	va_copy(argptr2, argptr);

